lib/fs.js

```diff
@@ -2009,12 +2009,15 @@ function ReadStream(path, options) {
 
   if (this.start !== undefined) {
     if (typeof this.start !== 'number') {
-      throw new ERR_INVALID_ARG_TYPE('start', 'number', this.start);
+      throw new ERR_INVALID_ARG_TYPE('options.start', 'number', options.start);
+    }
+    if (!Number.isInteger(this.start)) {
+      throw new ERR_OUT_OF_RANGE('options.start', 'an integer', options.start);
```

Comment:
Sorry, I guess I expressed myself not well enough in my last comment. What I meant is that you can move the type check inside of the `Number.isInteger` check. That way it will only do the type check in the negative case.

As in:

```js
if (!Number.isInteger(this.start)) {
  if (typeof this.start !== 'number') {
    throw new ERR_INVALID_ARG_TYPE(...);
  }
  throw new ERR_OUT_OF_RANGE(...);
}
```

Reply:
I think his approach is okay, maybe faster.

With
```js
if (!Number.isInteger(this.start)) {
  if (typeof this.start !== 'number') {
    throw new ERR_INVALID_ARG_TYPE(...);
  }
  throw new ERR_OUT_OF_RANGE(...);
}
```
, both `ERR_INVALID_ARG_TYPE` and `ERR_OUT_OF_RANGE` cases run 2 checks to be reached; while with
```js
if (typeof this.start !== 'number') {
  throw new ERR_INVALID_ARG_TYPE(...);
}
if (!Number.isInteger(this.start)) {
  throw new ERR_OUT_OF_RANGE(...);
}
```
, `ERR_INVALID_ARG_TYPE` runs 1 check to be reached, and `ERR_INVALID_ARG_TYPE` runs 2.